src/java.desktop/share/classes/javax/swing/text/DefaultEditorKit.java

```diff
@@ -642,6 +642,20 @@ public void write(Writer out, Document doc, int pos, int len)
      */
     public static final String endLineAction = "caret-end-line";
 
+    /**
+     * Name of the <code>Action</code> for moving the caret
+     * to the end of a line and logically upward one position.
+     * @see #getActions
```

Comment:
Tested the native behavior in XCode and it seems that this shortcut moves the cursor to the end if not at the end already or one line down otherwise.

Reply:
The code does that only...javadoc was wrong..fixed that